pkg/proxy/winkernel/proxier.go

```diff
@@ -926,6 +929,7 @@ func (proxier *Proxier) syncProxyRules() {
 
 	hnsNetworkName := proxier.network.name
 	hns := proxier.hns
+	cbr0HnsEndpoint, _ := hns.getEndpointByName("cbr0")
```

Comment:
cbr0 is GCP specific (https://kubernetes.io/docs/concepts/cluster-administration/networking/#google-compute-engine-gce), can this be generalized?

Reply:
It also looks like this can collide because there could be multiple endpoints with the same name.

Reply:
yes I don't think we should be putting GCP specific logic here. where is the current cbr0 endpoint being created?

Reply:
We do not usually apply policies to endpoints other than the one that is part of the service.

What is the actual scenario/data path are you trying to enable with this fix? 
Can you please describe? That would help us figure out what is the correct solution here.

Reply:
It is trying to let the GCE load balancer health check the nodes. The way GCE load balancer works is it doesn't do DNAT. Instead of getting `<node-ip>:<health-check-node-port>`, the node gets `<service-vip>:<health-check-node-port>`. so we need to do this to forward the health check package the kube-proxy's internal health check app. 

I have another PR open for further review: #99287. Let's move discussion there.